hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1500,11 +1499,9 @@ private void switchSnapshotCleanup(final boolean on) {
     try {
       snapshotCleanupTracker.setSnapshotCleanupEnabled(on);
       if (on) {
-        if (!getChoreService().isChoreScheduled(this.snapshotCleanerChore)) {
-          getChoreService().scheduleChore(this.snapshotCleanerChore);
-        }
+        getChoreService().scheduleChore(this.snapshotCleanerChore);
       } else {
-        getChoreService().cancelChore(this.snapshotCleanerChore);
+        this.snapshotCleanerChore.cancel();
```

Comment:
I know it's a pain but we need to have null check here

Reply:
I do not think it could be null here? This method is called inside the switchSnapshotCleanup method above and it could call synchronized on the snapshotCleanerChore instance...